.actions/setup_tools.py

```diff
@@ -42,33 +44,56 @@ def _load_py_module(name: str, location: str) -> ModuleType:
     return py
 
 
-def load_requirements(
-    path_dir: str, file_name: str = "base.txt", comment_char: str = "#", unfreeze: bool = True
-) -> List[str]:
-    """Loading requirements from a file.
+class _RequirementWithComment(pkg_resources.Requirement):
+    def __init__(self, *args: Any, comment: str = "", pip_argument: Optional[str] = None, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.comment = comment
+        assert pip_argument is None or pip_argument  # sanity check that it's not an empty str
+        self.pip_argument = pip_argument
+        self.strict = "# strict" in comment.lower()
+
+    def clean_str(self, unfreeze: bool) -> str:
+        # remove version restrictions unless they are strict
+        return self.project_name if unfreeze and not self.strict else str(self)
+
+
+def _parse_requirements(strs: Iterable) -> Iterator[_RequirementWithComment]:
+    """Adapted from `pkg_resources.parse_requirements` to include comments."""
+    lines = pkg_resources.yield_lines(strs)
+    pip_argument = None
+    for line in lines:
+        # Drop comments -- a hash without a space may be in a URL.
+        if " #" in line:
+            comment_pos = line.find(" #")
+            line, comment = line[:comment_pos], line[comment_pos:]
+        else:
+            comment = ""
+        # If there is a line continuation, drop it, and append the next line.
```

Comment:
do we have this anywhere?

Reply:
This is copy-pasted from the implementation of `pkg_resources.parse_requirements`

https://github.com/pypa/pkg_resources/blob/6f81a44010d1266494025647dd1e1f0befa5b26b/pkg_resources/__init__.py#L3022-L3041